assets/aml-benchmark/components/src/batch_benchmark_score/batch_score/header_handlers/header_handler.py

```diff
@@ -0,0 +1,58 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+"""The base class for header handlers.."""
+
+import json
+import os
+
+from abc import ABC, abstractclassmethod
+from ..utils.token_provider import TokenProvider
+from ..utils.common import constants
+
+class HeaderHandler(ABC):
+    """Class for header handler"""
+    def __init__(self,
+                 token_provider: TokenProvider,
+                 user_agent_segment: str = None,
+                 batch_pool: str = None,
+                 quota_audience: str = None,
+                 additional_headers: str = None) -> None:
+        if user_agent_segment and (":" in user_agent_segment or "/" in user_agent_segment):
+            raise Exception("user_agent_segment should not contain characters ':' or '/'")
```

Comment:
We need to make sure that the exceptions that we raise inherit from `BenchmarkUserException`, otherwise exceptions would be classified as `SystemError`. For example, please refer [this](https://github.com/Azure/azureml-assets/blob/280e56040035992cf7dee9c12e343ad17de95e71/assets/aml-benchmark/components/src/dataset_sampler/main.py#L210). This applies for all the exceptions that we raise in our code.

Reply:
changed